tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/editor/AbstractFreemarkerTest.java

```diff
@@ -268,4 +314,110 @@ public void emptyErrorLog() {
 		elv.deleteLog();
 		new WaitWhile(new JobIsRunning());
 	}
+	
+	/**
+	 * Checks whether executed freemarker template is resulting correct output
+	 * @param outputIsInFile boolean value that distinguishes if expected output is in String or will
+	 * be read from file
+	 * 
+	 * @param output expected output source (file or String)
+	 * @param alternateOutput alternate output string
+	 * @param runFile java file to be run as Java Application
+	 */
+	protected void checkFreemMarkerOutput(boolean outputIsInFile, String output, 
+			String alternateOutput, String... runFile) {
+		
+		String outputExpected = "";
+		if (outputIsInFile) {
+			String outputFilepath = getResourceAbsolutePath(
+					Activator.PLUGIN_ID, output);
+		
+			outputExpected = readOutputFromFile(outputFilepath);
+		} else {
+			outputExpected = output;
+		}
+
+		runJavaFile(runFile);
+		
+		ConsoleView cv = new ConsoleView();
+		cv.open();
+		try {
+			new WaitUntil(new ConsoleHasText(outputExpected), TimePeriod.NORMAL, false);
+		} catch (CoreLayerException exc) {
+			exc.printStackTrace();
+		}
+		String consoleText = cv.getConsoleText();
+		
+		if (!consoleText.equals(outputExpected)) {
+			log.error("Console text doesn't correspond with expected text");
+			log.dump("Console text:\n" + consoleText);
+			log.dump("Expected text:\n" + outputExpected);
+		}
+		// workaround for slightly different format on Windows
+		compareConsoleToOutput(consoleText, outputExpected, alternateOutput);
+	}
+	
+	/**
+	 * If given alternate expected result, checks whether we are running on Windows platform,
+	 * if yes, it checks whether console output contains alternate output. Otherwise it tries if 
+	 * expected output and resulting console output are equal, using String equals method.
+	 * Otherwise it checks whether trimmed console output is equal to trimmed expected output.
+	 * 
+	 * @param console String content of console that will be compared
+	 * @param output Expected output represented by String
+	 * @param alternateOutput Alternate output used for Windows platform
+	 */
+	private void compareConsoleToOutput(String console, String output, String alternateOutput) {
+		if (RunningPlatform.isWindows() && alternateOutput != null) {
```

Comment:
This seems like a logic error here (alternate output is either some string or empty string). I cannot see a situation where alternateOutput is null.